File: protect/src/main/java/com/webileapps/safeguard/KeyloggerDetection.java
package com.webileapps.safeguard;

import android.content.Context;
import android.provider.Settings;

public class KeyloggerDetection {
    public static boolean isAccessibilityServiceEnabled(Context context) {
        try {
            String enabledServices = Settings.Secure.getString(
                context.getContentResolver(),
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
            );
            if (enabledServices != null && enabledServices.contains(context.getPackageName())) {
                return true;
            }
        } catch (Exception ignored) {
        }
        return false;
    }
}

File: protect/src/main/java/com/webileapps/safeguard/KeyloggerDetection.kt
package com.webileapps.safeguard

import android.content.Context
import android.provider.Settings

object KeyloggerDetection {
    fun isAccessibilityServiceEnabled(context: Context): Boolean {
        try {
            val enabledServices = Settings.Secure.getString(
                context.contentResolver,
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
            )
            if (enabledServices != null && enabledServices.contains(context.packageName)) {
                return true
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return false
    }
}
